### ISSUE TITLE
[STORM-813] Storm-starter cannot run multilang topology with exec:java with new multilang packages

### ISSUE THREAD

**Issue by jira-importer** [apache/storm#4654]
<p>After <a href="https://issues.apache.org/jira/browse/STORM-748" title="Package Multi-Lang scripts so they don&#39;t have to be duplicated" class="issue-link" data-issue-key="STORM-748"><del>STORM-748</del></a>, storm-starter doesn't have multilang files now, and just having multilang artifact as dependency.<br/>
With supervisor it works like a charm, but now mvn exec:java with multilang topology doesn't work.</p>

<p>It would be better to remove broken functionality from README.markdown or make it work.</p>

---
<details><summary><i>Originally reported by <a title="Jungtaek Lim" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kabhwan">kabhwan</a>, imported from: <a href="https://issues.apache.org/jira/browse/STORM-813" target="_blank">Storm-starter cannot run multilang topology with exec:java with new multilang packages</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Jungtaek Lim" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kabhwan">kabhwan</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2015-06-09T20:52:33+01:00
<li><b>imported</b>: 2025-01-24
</ul></i>
</details>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>GitHub user HeartSaVioR opened a pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/548" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/548</a></p>

<p>    <a href="https://issues.apache.org/jira/browse/STORM-813" title="Storm-starter cannot run multilang topology with exec:java with new multilang packages" class="issue-link" data-issue-key="STORM-813"><del>STORM-813</del></a> Change storm-starter's README so that mvn exec:java cannot run multilang topology</p>

<p>    New multilang packages rely on supervisor deploy or maven dependency plugin.<br/>
    Therefore, storm-starter README should remove example which is not working now. </p>

<p>    Btw, actually storm-core shows that it could be possible to unpack multilang artifacts with maven plugin and run multilang topology with mvn exec:java, and I confirmed that can work with storm-starter.</p>

<p>    ```<br/>
    <plugin><br/>
<artifactId>maven-dependency-plugin</artifactId><br/>
<version>2.8</version><br/>
<executions><br/>
    <execution><br/>
<id>copy-dependencies</id><br/>
<phase>compile</phase><br/>
<goals><br/>
    <goal>copy-dependencies</goal><br/>
</goals><br/>
<configuration><br/>
    <overWriteReleases>false</overWriteReleases><br/>
    <overWriteSnapshots>false</overWriteSnapshots><br/>
    <overWriteIfNewer>true</overWriteIfNewer><br/>
    <includeScope>runtime</includeScope><br/>
</configuration><br/>
    </execution><br/>
    <br/>
    <execution><br/>
<id>unpack</id><br/>
<phase>process-resources</phase><br/>
<goals><br/>
    <goal>unpack</goal><br/>
</goals><br/>
<configuration><br/>
    <artifactItems><br/>
<artifactItem><br/>
    <groupId>org.apache.storm</groupId><br/>
    <artifactId>multilang-ruby</artifactId><br/>
    <version>${project.version}</version><br/>
</artifactItem><br/>
<artifactItem><br/>
    <groupId>org.apache.storm</groupId><br/>
    <artifactId>multilang-python</artifactId><br/>
    <version>${project.version}</version><br/>
</artifactItem><br/>
<artifactItem><br/>
    <groupId>org.apache.storm</groupId><br/>
    <artifactId>multilang-javascript</artifactId><br/>
    <version>${project.version}</version><br/>
</artifactItem><br/>
    </artifactItems><br/>
    <outputDirectory>${project.build.directory}/classes</outputDirectory><br/>
</configuration><br/>
    </execution><br/>
</executions><br/>
    </plugin><br/>
    ```</p>

<p>    If we prefer latter approach, I'll change my PR.</p>

<p>    Thanks in advance!</p>

<p>You can merge this pull request into a Git repository by running:</p>

<p>    $ git pull <a href="https://github.com/HeartSaVioR/storm" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/HeartSaVioR/storm</a> <a href="https://issues.apache.org/jira/browse/STORM-813" title="Storm-starter cannot run multilang topology with exec:java with new multilang packages" class="issue-link" data-issue-key="STORM-813"><del>STORM-813</del></a></p>

<p>Alternatively you can review and apply these changes as the patch at:</p>

<p>    <a href="https://github.com/apache/storm/pull/548.patch" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/548.patch</a></p>

<p>To close this pull request, make a commit to your master/trunk branch<br/>
with (at least) the following in the commit message:</p>

<p>    This closes #548</p>

<hr />
<p>commit a448be54543783225671eb7578b872c75645fe3e<br/>
Author: Jungtaek Lim <kabhwan@gmail.com><br/>
Date:   2015-05-11T14:21:23Z</p>

<p>    <a href="https://issues.apache.org/jira/browse/STORM-813" title="Storm-starter cannot run multilang topology with exec:java with new multilang packages" class="issue-link" data-issue-key="STORM-813"><del>STORM-813</del></a> notify mvn exec:java cannot run multilang topology</p>

<ul>
	<li>New multilang packages rely on supervisor deploy or maven dependency plugin</li>
</ul>


<hr />

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user ptgoetz commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/548#issuecomment-105627437" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/548#issuecomment-105627437</a></p>

<p>    +1</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user HeartSaVioR commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/548#issuecomment-108116665" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/548#issuecomment-108116665</a></p>

<p>    It would be better to include this to 0.10.0 cause multilang package is planned to be included to 0.10.0.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user asfgit closed the pull request at:</p>

<p>    <a href="https://github.com/apache/storm/pull/548" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/548</a></p>
